src/worldmap.cc

```diff
@@ -4188,9 +4195,9 @@ static bool wmGameTimeIncrement(int ticksToAdd)
 
     // SFALL: Fix Pathfinder perk.
     int pathfinderRank = perkGetRank(gDude, PERK_PATHFINDER);
-    double bonus = static_cast<double>(ticksToAdd) * static_cast<double>(pathfinderRank) * 0.25 + gGameTimeIncRemainder;
-    gGameTimeIncRemainder = modf(bonus, &bonus);
-    ticksToAdd -= static_cast<int>(bonus);
+    double newTicks = static_cast<double>(ticksToAdd) * (1.0 - static_cast<double>(pathfinderRank) * 0.25) * gScriptWorldMapMulti + gGameTimeIncRemainder;
+    gGameTimeIncRemainder = modf(newTicks, &newTicks);
+    ticksToAdd = static_cast<int>(newTicks);
```

Comment:
Are you sure there is no mistake here? Before gGameTimeIncRemainder was added to `bonus` and then bonus is subtracted from `ticksToAdd` (negative effect). Now it will be added instead (positive effect).

Reply:
Good question, and thanks for checking the math.

`gGameTimeIncRemainder` is accumulating the fractional component of ticks to make sure that int truncation is not resulting in zero effect for Pathfinder and similar boni.  Whether it's applies to `x` or to `y` (in `x = 1 - y`) is essentially equivalent.  There is a difference though: before it was applying to `y` so any fractional part would be applied a frame earlier than when it would with this formula.

I'm not sure which is strictly "right", since this is a sfall fix, not original code.  The way it is in this PR matches sfall (to my reading, anyway)